### DISCUSSION TITLE
How to select the bubble parameter value

### DISCUSSION THREAD

**Discussion by Lamed-git** [ttadano/alamode#173] [General]
Dear Terumasa Tadano,
I find in the source code that bubble can take the values 0,1,2,3, and I want to know if the different values make the calculation of the bubble process different or do they involve different processes? For example, what are the differences between these three files, and how should we choose the appropriate bubble parameter value?
![image](https://github.com/ttadano/alamode/assets/81789705/13cf6cea-c3ed-4151-8a4d-19846d8a6486)
I noticed that your previous article mentioned a similar process, but it's not quite clear how the code is implemented.
![image](https://github.com/ttadano/alamode/assets/81789705/ba66b778-1963-4589-b881-21069fcb57d1)
Looking forward to your reply!

**Comment by ttadano** [comment]
All `BUBBLE > 0` options involve the same process, i.e., the diagram (b) shown in that attached figure.

The key difference is the way they estimate the quasiparticle phonon frequencies. More specifically, when computing the frequency shift associated with the bubble self-energy, $\Sigma^{B}(\omega)$, within the quasiparticle approximation, we approximate the $\omega$-dependent $\Sigma^{B}(\omega)$ by a representative value at a specific $\omega$, say $\omega=\omega_0$, as $\Sigma^{B}(\omega)\approx \Sigma^{B}(\omega_0)$.

The choice for $\omega_0$ is not unique. 
- `BUBBLE = 1` sets $\omega_0 = 0$ (static approximation)
- `BUBBLE = 2` sets $\omega_0 = \omega_{q}^{S}$ (The quasiparticle energy within SCPH without bubble correction)
- `BUBBLE = 3` self-consistently determine $\omega_0$ so that $\omega_0$ equals to the quasiparticle energy after considering the bubble correction.

`BUBBLE = 3` is the most advanced, at least theoretically, but it is not clear if this option always gives a better result.

For each `BUBBLE > 0` option, we change the file suffix so that they can be easily distinguished.

**Reply by Lamed-git** [reply]
Thanks for your reply, now I can understand the meaning of this parameter, I will seriously test my own system, thank you very much!
